src/LodeRunner.API/src/Core/NgsaLog.cs

```diff
@@ -72,29 +73,37 @@ public class NgsaLog
         /// <param name="method">method to log.</param>
         /// <param name="message">message to log.</param>
         /// <param name="context">http context.</param>
-        /// <param name="dictionary">optional dictionary.</param>
-        public void LogInformation(string method, string message, HttpContext context = null, Dictionary<string, object> dictionary = null)
-        {
-            if (LogLevel <= LogLevel.Information)
-            {
-                WriteLog(LogLevel.Information, this.GetDictionary(method, message, LogLevel.Information, null, context, dictionary));
-            }
-        }
-
+        /// <param name="dictionary">optional dictionary.</param>
+        /// <returns>The Task.</returns>
+        public async Task LogInformation(string method, string message, HttpContext context = null, Dictionary<string, object> dictionary = null)
+        {
+            await Task.Run(() =>
+            {
+                if (LogLevel <= LogLevel.Information)
+                {
+                    WriteLog(LogLevel.Information, GetDictionary(method, message, LogLevel.Information, null, context, dictionary));
+                }
+            });
+        }
+
         /// <summary>
         /// Log warning.
         /// </summary>
         /// <param name="method">method to log.</param>
         /// <param name="message">message to log.</param>
         /// <param name="eventId">Event ID.</param>
         /// <param name="context">http context.</param>
-        /// <param name="dictionary">optional dictionary.</param>
-        public void LogWarning(string method, string message, LogEventId eventId = null, HttpContext context = null, Dictionary<string, object> dictionary = null)
-        {
-            if (LogLevel <= LogLevel.Warning)
-            {
-                WriteLog(LogLevel.Warning, this.GetDictionary(method, message, LogLevel.Warning, eventId, context, dictionary));
-            }
+        /// <param name="dictionary">optional dictionary.</param>
+        /// <returns>The Task.</returns>
+        public async Task LogWarning(string method, string message, LogEventId eventId = null, HttpContext context = null, Dictionary<string, object> dictionary = null)
+        {
+            await Task.Run(() =>

```

Comment:
Looking at the code we could move the await Task.Run() to the WriteLog method and the return that from the LogInfo, LogWarn, and LogError methods.

Reply:
good call,! fixed.